server/src/main/java/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java

```diff
@@ -1238,6 +1242,11 @@ protected void runInContext() {
                 s_logger.info("Found " + routers.size() + " running routers. Fetching, analysing and updating DB for the health checks.");
 
                 for (final DomainRouterVO router : routers) {
+                    if (!RouterHealthChecksEnabled.valueIn(router.getDataCenterId())) {
+                        s_logger.debug("Skipping fetching of router health check results as its disabled for router " + router.getUuid());
+                        continue;
+                    }
```

Comment:
should we not log the reverse? when it is enabled?

Reply:
I think we have that covered @DaanHoogland . We log on line 1242 that we are fetching and analysing results. Then if we are skipping that is logged. Any subsequent steps are logged as well in sub-methods.

Reply:
don't forget a lot of prod installs have debug enabled. but than again they should just patch the levels and send PRs in for that ;)
What is it you need (as developer or otherwise) so you want to see this?
I would think if you want to see this you want to see bot enabled and disabled. You say enabled is covered but those loggings might be missed because of exceptions.

Also, this is why i kept asking about valueIn(zone). Did you forget to remove this one?

Reply:
It's removed in the 2nd commit, Daan. There are 2 commits after your last review.